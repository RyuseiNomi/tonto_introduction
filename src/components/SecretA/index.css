.secret-a-background {
  background-color: black;
  padding: 300px 0 300px 0;
}

.secret-a-button {
  color: red;
}
