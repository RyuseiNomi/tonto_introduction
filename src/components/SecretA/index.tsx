import React from 'react';
import './index.css';

type Props = {
  toggleScroll: Function;
}

function SecretA(props: Props) {

  const { toggleScroll } = props;
  const scrollToTop = () => {
    window.scrollTo({
      top:0,
      behavior: "smooth",
    });
  }
  const toggleEvents = () => {
    toggleScroll("up");
    scrollToTop();
  }

  return (
    <div className="secret-a-background">
      <button
        className="secret-a-button"
        data-testid="secret-a-button"
        type="button"
        onClick={() => toggleEvents()}
      >
      ｴ  ﾝ ｹﾞ ｲﾌﾞ       ﾆ   ﾅﾙ?
      </button>
    </div>
  );
}
export default SecretA;
