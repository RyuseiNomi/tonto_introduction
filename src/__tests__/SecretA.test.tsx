import { render } from 'react-dom';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import SecretA from './../components/SecretA';

let container = null;
let root = null;

beforeEach(() => {
  // container の定義
  container = document.createElement("div");
  document.body.appendChild(container);
  root = createRoot(container);
});

afterEach(() => {
  // 定義した contaienr の除去
  root.unmount(container);
  container.remove();
  container = null;
});

// =============== snapshot test =================
it('SecretA コンポーネントが表示されること', () => {
  // スクロール矢印の向きを制御するダミーの関数
  const toggleScroll = jest.fn();
  const component = renderer.create(<SecretA toggleScroll={toggleScroll} />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

// ================ act tests ====================
it('ボタンが 1 つ存在していること', () => {
  // スクロール矢印の向きを制御するモック関数
  const toggleScroll = jest.fn();

  act(() => {
    root.render(<SecretA toggleScroll={toggleScroll} />)
  });

  expect(document.querySelector("[data-testid=secret-a-button]")).toBeTruthy();

});

it('ボタンをクリックした時、スクロールの向きが切り替わる関数が呼ばれること', () => {
  // スクロール矢印の向きを制御するモック関数
  const toggleScroll = jest.fn();

  act(() => {
    root.render(<SecretA toggleScroll={toggleScroll} />)
  });

  expect(toggleScroll).toHaveBeenCalledTimes(0);

  const button = document.querySelector("[data-testid=secret-a-button]");
  act(() => {
    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  })

  expect(toggleScroll).toHaveBeenCalledTimes(1);
});
