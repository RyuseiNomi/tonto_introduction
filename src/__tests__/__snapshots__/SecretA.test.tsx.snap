// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecretA コンポーネントが表示されること 1`] = `
<div
  className="secret-a-background"
>
  <button
    className="secret-a-button"
    data-testid="secret-a-button"
    onClick={[Function]}
    type="button"
  >
    ｴ  ﾝ ｹﾞ ｲﾌﾞ       ﾆ   ﾅﾙ?
  </button>
</div>
`;
